lib/puppet/pal/pal_impl.rb

```diff
@@ -176,13 +176,21 @@ def self.with_catalog_compiler(
       end
     end
 
+    # We need to make sure to set these back when we're done
+    previous_tasks_value = Puppet[:tasks]
+    previous_code_value = Puppet[:code]
```

Comment:
This feels like it should be a context override block. But I don't see an equivalent for settings, or know if moving them to the context at some point would even be a good idea.

Reply:
Yeah I tried using a context override and it definitely doesn't work for settings. I think this might be the best we can do for now.